tests/phpunit/CRM/Utils/SystemTest.php

```diff
@@ -35,4 +37,73 @@ public function testEvalUrl() {
     $this->assertEquals('http://example.com/?cms=UnitTests', CRM_Utils_System::evalUrl('http://example.com/?cms={uf}'));
   }
 
+  /**
+   * Test the redirect hook.
+   *
+   * @param string $url
+   * @param array $parsedUrl
+   *
+   * @dataProvider getURLs
+   */
+  public function testRedirectHook($url, $parsedUrl) {
+    $this->hookClass->setHook('civicrm_alterRedirect', array($this, 'hook_checkUrl'));
```

Comment:
This confused me a little while reading -- in the vast majority of test classes, a function named `hook_foo` is an implementation of `hook_foo`. Here the names don't match up. Suggestion: https://gist.github.com/totten/b5570ae6062fc115c4fb5e300385543b

Other examples:

```
tests/phpunit/api/v3/ContributionPageTest.php:  public function hook_civicrm_alterPaymentProcessorParams($paymentObj, &$rawParams, &$cookedParams) {
tests/phpunit/api/v3/MembershipTest.php:  public function hook_civicrm_pre_update_create_membership($op, $objectName, $id, &$params) {
tests/phpunit/api/v3/SelectQueryTest.php:  public function hook_civicrm_selectWhereClause($entity, &$clauses) {
tests/phpunit/Civi/Angular/ManagerTest.php:  public function hook_civicrm_alterAngular($angular) {
tests/phpunit/Civi/Angular/ManagerTest.php:  public function hook_civicrm_angularModules_fooBar(&$angularModules) {
tests/phpunit/Civi/CCase/SequenceListenerTest.php:  public function hook_caseTypes(&$caseTypes) {
tests/phpunit/Civi/Core/Event/GenericHookEventTest.php:  public function hook_civicrm_ghet(&$roString, &$rwString, &$roArray, &$rwArray, $plainObj, &$refObj) {
tests/phpunit/Civi/Test/ExampleHookTest.php:  public function hook_civicrm_alterContent(&$content, $context, $tplName, &$object) {
tests/phpunit/CiviTest/CiviCaseTestCase.php:  public function hook_caseTypes(&$caseTypes) {
tests/phpunit/CRM/ACL/ListTest.php:  public function hook_civicrm_aclWhereClause($type, &$tables, &$whereTables, &$contactID, &$where) {
tests/phpunit/CRM/Case/BAO/CaseTypeForkTest.php:  public function hook_caseTypes(&$caseTypes) {
tests/phpunit/CRM/Contribute/Form/Task/PDFLetterCommonTest.php:  public function hook_tokens(&$tokens) {
tests/phpunit/CRM/Contribute/Form/Task/PDFLetterCommonTest.php:  public function hook_aggregateTokenValues(&$values, $contactIDs, $job = NULL, $tokens = array(), $context = NULL) {
tests/phpunit/CRM/Core/FieldOptionsTest.php:  public function hook_civicrm_fieldOptions($entity, $field, &$options, $params) {
tests/phpunit/CRM/Group/Page/AjaxTest.php:  public function hook_civicrm_aclGroup($type, $contactID, $tableName, &$allGroups, &$currentGroups) {
tests/phpunit/CRM/Mailing/MailingSystemTest.php:  public function hook_alterMailParams(&$params, $context = NULL) {
tests/phpunit/CRM/Utils/SystemTest.php:  public function hook_civicrm_alterRedirect($urlQuery, $context) {
```